pkg/manager/protodep.go

```diff
@@ -0,0 +1,42 @@
+package manager
+
+import (
+	"context"
+
+	"github.com/solo-io/protodep/protodep"
+)
+
+type DepFactory interface {
+	Ensure(ctx context.Context, opts *protodep.Config) error
```

Comment:
nit: `depFactory.Ensure` doesn't seem like a clear API to me. how does this relate to how `protodep` is used? i assume this is the API called by the user?